### ISSUE TITLE
Map orientation and Display position can become decoupled (forever)

### ISSUE THREAD

**Issue by sonora** [osmandapp/OsmAnd#16157]
We create this issue by now explicitly exposing the option "Display position" in the settings: If after a fresh install you have never directly manipulated that option, instead always only change the option "Map orientation" to your liking, the "Display position" automatically follows suit in the only way which makes sense logically:
- Display position is "Center," in all isotropic modes,
- and "Bottom" only in mode "Movement direction" (the only mode where you have a discrete direction --your movement direction--for which it makes sense to reserve more screen real estate than for others.

When playing around to test #16112, I noticed that if you manually override the "Display position" only once, that logic becomes decoupled forever, in all profiles, and without a way to reinstate it: You will now have to set "Display position" and "Map orientation" always explicitly.

We could either
- (1) hide the setting "Display position" from the UI and always apply the automatic logic that it is "Bottom" for "Movement direction",  "Center" in all other cases. (Anybodyknows a use case why you would want to deviate?)
- (2) or we need some way to reset the "Display position" back to the "Automatic" option OsmAnd provides out of the box. The easiest way to achieve this is that changing "Map orientation" always overrides "Display position" with the automatic value, so if you want to deviate you have to explicitly set a deviating "Display position" after you changed the "Map orientation" setting.

**Comment by pebogufi** [comment]
Prefer (1), don't see use cases, which recommend different behavior.

**Comment by mikehgentry** [comment]
You have a direction in 'Compass direction' mode as well, surely.

**Comment by vshcherb** [comment]
We need to make explicit cause it causes lots of question, so we can add new entry Display Position - "Automatic depending on Map orientation"

**Comment by sonora** [comment]
@vshcherb Here's the PR: #16267.

![Screenshot_20230117-235224_OsmAnd~](https://user-images.githubusercontent.com/845735/213034660-22d6869f-2af5-4481-9af0-4c52a9f590be.jpg)

**Comment by vshcherb** [comment]
We think to add more description to UI to explain what's going on behind Automatic

**Comment by Nine-Tailed** [comment]
I would also like an option to set the orientation when navigation is turned on, use the current or set one, for example, in the movement direction. After navigation is complete, return to the previous orientation.

**Comment by sonora** [comment]
@Nine-Tailed That's what the profiles are for: Use or create one for map viewing, another one for navigation, each with the settings of your liking.

**Comment by sonora** [comment]
@vshcherb Text suggestion: If desired, could use a new
```
<string="position_placement_automatic_descr">Bottom for movement direction, center for all other map orientations.</>
```
as description under option Automatic.

**Comment by sonora** [comment]
PS: This radio button general profile setting still uses a dialog, most others have been converted to a bottom sheet. I gues that's the goal here as well?

**Comment by EugeneZmeuk** [comment]
https://osmand.net/docs/user/personal/profiles#appearance

https://osmand.net/docs/user/map/interact-with-map#compas

**Comment by sonora** [comment]
Ok, thanks! That requires an initiative beyond just the scope of this issue
- to migrate everything to using a bottom_sheet,
- and upgrade our bottom_sheets to the design proposed also for https://github.com/osmandapp/OsmAnd/issues/15836#issuecomment-1385619073, i.e. displaying icons.

Maybe we should make this a separate story "_Harmonize pref setting UI_" and separate from these single functional Issues?

**Comment by dmpr0** [comment]
* I propose to rename **Position placement** to **Location position on screen**. 
* *  For example, Apple use **[Position on screen](https://cdn.osxdaily.com/wp-content/uploads/2013/10/change-mac-dock-location.jpg)** in Dock setting, Microsoft use **[Taskbar location on screen](https://www.isumsoft.com/images/windows-10/change-taskbar-position-on-windows-10-screen/change-taskbar-location-on-screen.png)** for Taskbar settings.
* * Use new icon `ic_action_display_position_auto`, [link](https://github.com/osmandapp/OsmAnd/commit/d86a2d6d0990b922296940f6cc0c2680c905c3f6)
* Change a little bit proposed description `Depends on Map orientation, bottom for movement direction, center for all other.`
<table style="width:100%">
  <tr>
  </tr>
  <tr>
    <td><img src="https://user-images.githubusercontent.com/9365308/214342939-d2ae3617-2e9e-4492-ae32-8295ac393528.png"  width="300" height="auto"/> </td>
  </tr>
</table>

**Comment by sonora** [comment]
Yes, I am fine with everything you say, but have this remark regarding your first bullet::

In OsmAnd we use "_Position_" for where you are ("the blue dot"), "_Location_" for any other geo-location, e.g. a favorite, POI, or a destination.

If we go to "Location Position on map", it's rather confusing because we now use "Location" for what elsewhere we call "Position", and now use "Position" in a third meaning, namely where things are placed on the device screen. :wink:

To work around this, I had come up with "_Position placement_" meaning: "Where to we place the MyPosition marker o the screen". It's also nice and short. But if we wanted we could expand to "MyPosition placement".

(PS: Reason why I changed the old "Display Position" in the first place is that that was ambiguous: It was meant to mean "Show Position at". But some people read "Display" as a noun, which changes the meaning completely, because now it does not talk about where we place the widget, but where we place the device display.... )

**Comment by mikehgentry** [comment]
"Focus placement" or "Display focus" or something similar is another option

**Comment by vshcherb** [comment]
@sonora we can't close it, we need to make bottom sheet with explanation otherwise it will be always asking question what automatic means.

**Comment by sonora** [comment]
I was joking... :wink:

**Comment by mikehgentry** [comment]
"Map focus"

seems pretty unambiguous to me?

**Comment by dmpr0** [comment]
> @dmpr0: I opened this Issue. Can I now vote it solved? 😉 Now seriously: Yes, I am fine with everything you say, but have this remark regarding your first bullet::
> 
> In OsmAnd we use "_Position_" for where you are ("the blue dot"), "_Location_" for any other geo-location, e.g. a favorite, POI, or a destination.
> 
> If we go to "Location Position on map", it's rather confusing because we now use "Location" for what elsewhere we call "Position", and now use "Position" in a third meaning, namely where things are placed on the device screen. 😉
> 
> To work around this, I had come up with "_Position placement_", meaning: "Where we place the MyPosition marker on the screen". It's also nice and short. But if we wanted we could expand to "MyPosition placement".
> 
> (PS: Reason why I changed the old "Display Position" in the first place is that that was ambiguous: It was meant to mean "Show Position at". But some people read "Display" as a noun, which changes the meaning completely, because now it does not talk about where we place the widget, but where we place the device display.... )

![Screenshot 2023-01-25 at 09 28 26](https://user-images.githubusercontent.com/9365308/214504483-9a8d24a5-b47e-4c2f-9d86-e294affb1da0.png)

![Screenshot 2023-01-25 at 09 28 45](https://user-images.githubusercontent.com/9365308/214504521-e46261df-8cfc-4bcd-8474-456d88954d3d.png)

**Comment by sonora** [comment]
Yes, I think "_Map focus_" nails it ! It also goes well with "_Map orientation_", the option displayed immediately above it under the general profile options. I vote for that!

**Comment by dmpr0** [comment]
**Map focus** doesn't have any connection with function. I propose to restore all changes to two options, until we decide how to deal with changes like this.

**Comment by sonora** [comment]
I don't understand

> Map focus doesn't have any connection with function.

because when you look at the list on the "General profile settings" screen, it starts with

- _App theme_
- _Map orientation_
- _Map focus_ (or _Position placement_)
- _Screen orientation_
- _Screen control_
- ... and so on.

All of these are _noun-noun_ items, and linguistically fit perfectly together in this scheme?

**Comment by seahorse41** [comment]
Why not "Map center" ?

**Comment by sonora** [comment]
@vshcherb As a first step I have prepared https://github.com/osmandapp/OsmAnd/pull/16329., it achieves using the correct icons on the parent screen and the standard bottom sheet:

![Screenshot_20230126-145230_OsmAnd~](https://user-images.githubusercontent.com/845735/214868443-f11ea89a-bad8-40fa-b3d7-eebfb46e26c6.jpg)

- [ ] What still needs to be done is adjust the standard bottom sheet (for all settings) to display the icons as well.

**Comment by GPSFietser** [comment]
I would like to coin the phrase **Look ahead** instead of **Display position**. (Placing the map off center so it shows more of the map in the direction you are going to has the function of ... _looking ahead_.)

Also, since the orientation settings are really important I think they deserve a separate sub menu: **Configure profile » Orientation**. This makes it easier to find, since you would not immediately expect orientation settings 'scattered' under General settings. The sub menu **Orientation** could hold the following settings:

1. **Default map _rotation_** (North up, Movement direction, Compass direction, Manual)
2. **Look ahead**: _select for each rotation mode_
	**North up** (Movement, Compass, Off)*
	**Movement direction** (On, Off)
	**Compass direction** (On, Off)
	**Manual rotation** (On, Off)
4. **2.5D view** (On, Off)
	Only useful when rotated in my opinion, so no 2.5D in North Up! 
6. **Screen orientation** (Portret, Landscape, Same as device)

\* **Look ahead** in **North Up** explained/feature request!: when the map orientation is _North up_ the map is placed off centre in such a way that you are looking ahead in the map. This means the position arrow is placed off centre along an (elliptic) circle depending on the compass or movement direction. This works really great when you prefer navigating using _North Up_. The app Pocket Earth (iOS only) has this feature, so you could try it out.

BTW: Can someone explain the option _Bottom_? It seems to me that _Auto_ and _Center_ is sufficient. When orientation is set to North Up the option _Bottom_ will place the position at the bottom, which is very weird when going in a southerly direction! As a cyclist I like to switch often between _Movement direction_ and _North Up_ and with the Bottom behaviour this is not working for me anymore since North Up is useless as soon as I go in a southerly direction.

**Comment by mikehgentry** [comment]
> BTW: Can someone explain the option Bottom? It seems to me that Auto and Center is sufficient. When orientation is set to North Up the option Bottom will place the position at the bottom, which is very weird when going in a southerly direction! As a cyclist I like to switch often between Movement direction and North Up and with the Bottom behaviour this is not working for me anymore since North Up is useless as soon as I go in a southerly direction.

Compass direction plus Bottom placement is of use on a bike - you can just point yourself in the direction you are thinking about going and get a good view of the path ahead without having to fiddle with zoom / pan (not always trivial wearing gloves with your phone in a bag / on a mount).

I used to just use the movement threshold for this: if you were moving it'd be in Movement direction mode, and if you stopped it'd revert to Compass direction. I would have had Display position set to Bottom I think, but if Automatic remained at the bottom when temporarily switching to Compass that'd cover my use case without an explicit Bottom option (unfortunately it's a moot point at the moment, because the threshold mode doesn't seem to be working for me).

**Comment by sonora** [comment]
The purpose of this issue was to facilitate going back to  a setting which is our installation default but could not be returned to once changed. That solved, it has now evolved to improve wording and UI a bit.

I personally quite like the novel idea of a "Look ahead" concept, and think it may deserve its own separate issue to be discusssed thoroughly.  In particular how it could be implemented with a degree of complexity most users can readily understand.  (E g. a 3-tiered hierarchy _App profile > Map orientation mode > Look ahead mode_ may be overdoing it, perhaps 2 tiers are sufficient.)

**Comment by vshcherb** [comment]
I upvote proposed @dmpr0 solution cause I'm pretty most of persons will understand including native / non-native speakers. All people feel that with start app location is focused in the center of the map even if they don't express. And answering where do you want to see yourself on the map - bottom or center, I think proposed "Location position on screen" is totally in that paradigm.

**Comment by sonora** [comment]
The only issue I see with that wording is consistency: So far we have used "location" in the broader sense, but "position" for where you are (and in particular to reference our blue "Position" icons).  Our string `make_osmand_better_descr `illustrates this by saying "_No data about your position or locations you view on the map ..._".

 Here are some examples in code (and this extends to documentation, of course:):

    <string name="my_location">My position</string>
    <string name="select_navigation_icon">Position icon while moving</string>
    <string name="select_map_icon">Position icon at rest</string>
    <string name="snap_to_road_descr">Current position icon will snap to the current navigation route</string>
    <string name="navigate_to_track_descr">Navigate from my position to the track</string>
    <string name="search_my_location">Find my position</string>
    <string name="simulate_your_location">Simulate your position</string>
    <string name="shared_string_my_location">My Position</string>
    <string name="unknown_location">Position not yet known.</string>
    <string name="back_to_location">Return to position</string>
    <string name="trip_recording_logging_interval_info">The logging interval sets the time period in which OsmAnd will request the current position data.</string>
    <string name="monitoring_min_accuracy_descr">This will record only points measured with a minimum "accuracy" indication (in meters/feet, as reported by Android for your chipset). Accuracy is the proximity of measurements to the true position, and is not directly related to precision, which is the scatter of repeated measurements.</string>
    <string name="make_osmand_better_descr">Allow OsmAnd to collect and process anonymous app usage data. No data about your position or locations you view on the map are collected.\n\nConfigure any time in \'Settings\' → \'Privacy and Security\'.</string>
    <string name="show_guide_line_descr">Show directional line from your position to the active marker locations.</string>
    <string name="add_location_as_first_point_descr">Add your position as point of departure to plan the perfect route.</string>
    <string name="animate_my_location_desc">Turn on animated map panning of \'My position\' during navigation.</string>
    <string name="simulate_your_location_stop_descr">Stop simulating your position.</string>
    <string name="simulate_your_location_descr">Simulate your position using a calculated route or a recorded GPX track.</string>
    <string name="simulate_your_location_gpx_descr">Simulate your position using a recorded GPX track.</string>
    <string name="save_track_to_gpx_globally_descr">General position logging to a GPX file can be turned on or off using the GPX logging widget on the map.</string>
    <string name="enable_plugin_monitoring_services">Enable the \"Trip recording\" plugin to use position logging services (GPX logging, online tracking)</string>

**Comment by GPSFietser** [comment]
I do understand the concept of placing the gps position at the bottom, but it only makes sense when the map is rotated in movement or compass direction! 

The current implementation also places the gps position at the bottom in North Up (fixed), and this is really not working in _North Up_ when going in southerly directions!

So, my point is that when in _North Up_ the gps position should always be placed in the center of the map. (Or use a circular offset as I proposed, but that is another discussion)

**Location position on screen** (already in the iOS beta):

- _Auto_: _movement_ placed at bottom, _compass_ and north up placed in the center.
- _Bottom_: _movement_ and _compass_ at the bottom (keep _north up_ in the center!)
- _Center_: always in the center

The options are very confusing since it is not immediately clear what is intended. That's why Look ahead might be a solution. 

**Look ahead**:

- _Off_ (or _Center_)
- _Movement direction_
- _Movement direction_ & _Compass_
- _All orientations_ (offset in a circle around the center when in North Up)

![Schermafbeeldingen - 1 van 1](https://user-images.githubusercontent.com/122881440/217779907-7ab3726a-f47a-47be-b254-2a1e0e746842.jpeg)

**Comment by jsgblom** [comment]
I'm a long time user and a big fan of OsmAnd and I am using it on both iOS and Android (and some other navigation apps, mostly Pocket Earth and MapOut on iOs, but also Komoot and Wahoo element bolt). My main use is mountain biking, long distance cycling and hiking. I give navigation courses using OsmAnd in the Netherlands to long distance cyclists and hikers.

I would like to take the discussion in Map orientation further but since I'm new to github and the osmand discussions I could use a hand. Maybe also some guidelines on how to go about issues/discussions that are related to both iOS and Android.

(btw: I changed my account name from @GPSFietser to @jsgblom)

**Comment by jsgblom** [comment]
I just updated the Android release version (4.3.12) and the change of the profile setting **Display position** -> **Location position on screen**  is not solved/implemented. There is no option _Automatic_.  (In the iOS beta version it is already implemented as discussed here.) 

North Up is not workable when placed at the bottom, so users who want to switch regularly between North Up and Movement direction will have a problem. So, I would consider this as a bug and give it high priority.

**Comment by Zirochkabila** [comment]
OsmAnd~ 4.4.0#14189mqta, Released: 2023-02-20
Done

**Comment by Zirochkabila** [comment]
@jsgblom Try downloading the nightly version - it is implemented there
You can find "Automatic" option in this way: Menu > Configure profile > General settings > Display position

**Comment by DmitryAlexei** [comment]
https://osmand.net/docs/user/personal/profiles#appearance
